### PR TITLE
HY-4966 Strong mode and 2 lints

### PR DESCRIPTION
**Pull request by robbecker-wf**
This turns on strong mode and 2 lints to check for closed streams.

# Testing
 - [ ] Pull this into some larger apps like the wdesk example app or GRC and make sure routing works and no runtime type errors

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Raven
Number of Findings: 0

File: .travis.yml [hunk 1]
```diff
@@ -9,5 +9,3 @@ script:
   - pub run dart_dev format --check
   - pub run dart_dev analyze
   - pub run dart_dev test
-  - pub run dart_dev coverage --no-html
-  - bash <(curl -s https://codecov.io/bash) -f coverage/coverage.lcov
```
Comment (robbecker-wf) [inline .travis.yml:13]: coverage would hang forever in travis for some reason, so I finally just had to leave it off

**Comment by robbecker-wf** [comment]
I pulled this into GRC and smoke tested changing routes and such.
route changes

File: .travis.yml (hunk 1 above)
Comment (aaronstgeorge-wf) [reply .travis.yml:13]: maybe codecov was just down?

Comment (robbecker-wf) [reply .travis.yml:13]: no it would hang on line 12 trying to run coverage even before trying to upload results to codecov

**Review by trentgrover-wf — APPROVED** [review]
approve of all of rob's commits

**Comment by trentgrover-wf** [comment]
@robbecker-wf I added a commit because a couple of tests were failing for me locally on dart 1.24.1 (just a couple of async / await tweaks in the tests) and the new test package had some deprecations to address

can you +1 my commit?

File: test/providers/common_tests.dart [hunk 2]
```diff
@@ -282,12 +282,12 @@ commonProviderTests(RouterFactory routerFactory) {
   });
 
   group('hierarchical routing', () {
-    void _testParentChild(Pattern parentPath, Pattern childPath,
+    Future<bool> _testParentChild(Pattern parentPath, Pattern childPath,
```
Comment (robbecker-wf) [inline test/providers/common_tests.dart:285]: shouldn't a function that returns Future be marked async ?

File: test/providers/common_tests.dart [hunk 3]
```diff
@@ -717,7 +717,7 @@ commonProviderTests(RouterFactory routerFactory) {
       expect(router.activePath.map((r) => r.name), ['foo']);
     });
 
-    void _testAllowLeave(bool allowLeave) {
+    Future<bool> _testAllowLeave(bool allowLeave) {
```
Comment (robbecker-wf) [inline test/providers/common_tests.dart:720]: same .. async keyword here?

File: test/providers/common_tests.dart (hunk 2 above)
Comment (trentgrover-wf) [reply test/providers/common_tests.dart:285]: only needs to be marked async if you're using await within the method
don't have to if you're directly returning the future

**Pull request closed without merging**